src/main/java/com/spring/demo_relationship/service/user/UserServiceImpl.java

```diff
@@ -36,29 +31,45 @@ public UserEntity getProfile(String email) {
 
     @Override
     public UserEntity updateProfile(UserEntity userEntity) {
+        UserEntity existingUser = getCurrentUser();
         existingUser.updateProfile(userEntity);
         userRepository.save(userEntity);
         return existingUser;
     }
 
     @Override
     public DoctorProfile updateProfile(DoctorProfile doctorProfile) {
+        UserEntity existingUser = getCurrentUser();
         DoctorProfile existingDoctorProfile = doctorProfileRepository.getDoctorProfileByDoctor(existingUser);
         existingDoctorProfile.updateDoctorProfile(doctorProfile);
         return doctorProfileRepository.save(existingDoctorProfile);
     }
 
     @Override
     public List<UserEntity> getAllUsersByRole(String role) {
-        return List.of();
+        Role roleEnum = Role.valueOf(role.toUpperCase());
+        return userRepository.findUsersByRole(roleEnum);
+    }
+
+    @Override
+    public List<UserEntity> getAllDoctors(String firstName, String lastName, String city) {
+        return userRepository.getDoctors(firstName,lastName,city);
```

Comment:
Use Criteria Queries search ( Criteria Queries in spring )
I think it's the best choice here 
try pagination not List<T>, Page<T> using Pageable<T>